ethcore/src/client/client.rs

```diff
@@ -426,17 +426,22 @@ impl<V> BlockChainClient for Client<V> where V: Verifier {
 		block.try_seal(self.engine.deref().deref(), seal)
 	}
 
+	fn engine(&self) -> &Engine {
+		self.engine.deref().deref()
+	}
+
 	// TODO [todr] Should be moved to miner crate eventually.
 	fn prepare_sealing(&self, author: Address, gas_floor_target: U256, extra_data: Bytes, transactions: Vec<SignedTransaction>)
-		-> Option<(ClosedBlock, HashSet<H256>)> {
+		-> (Option<ClosedBlock>, HashSet<H256>) {
```

Comment:
In the caller, we always want to know which transactions were invalid, even if there's no new block to mine.